### DISCUSSION TITLE
Anyone help me please// error: failed to build archive: Unknown attribute kind (86) (Producer: 'LLVM17.0.2' Reader: 'LLVM 15.0.6-rust-dev')     error: aborting due to previous error  10:37:23 ninja failed with: exit status 1

### DISCUSSION THREAD

**Discussion by Thereache** [community/community#103016] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

FAILED: /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib                                                                                        OUT_DIR=out ANDROID_RUST_VERSION=1.68.0 CARGO_CRATE_NAME=ring CARGO_PKG_VERSION=0.17.0-alpha.11 AR=prebuilts/clang/host/linux-x86/clang-r487747c/bin/llvm-ar prebuilts/rust/linux-x86/1.68.0/bin/rustc -C linker=build/soong/scripts/mkcratersp.py --emit link -o /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib --emit dep-info=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d.raw external/rust/crates/ring/src/lib.rs --extern libc=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/libc/liblibc/android_arm64_armv8-2a_rlib_rlib-std_apex31/liblibc.rlib --extern once_cell=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/once_cell/libonce_cell/android_arm64_armv8-2a_rlib_rlib-std_apex31/libonce_cell.rlib --extern spin=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/spin/libspin/android_arm64_armv8-2a_rlib_rlib-std_apex31/libspin.rlib --extern untrusted=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/untrusted/libuntrusted/android_arm64_armv8-2a_rlib_rlib-std_apex31/libuntrusted.rlib --extern std=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd/android_arm64_armv8-2a_rlib_apex31/libstd.rlib -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcompiler_builtins.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcore.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/librustc_demangle.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/liblibc.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcfg_if.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/liballoc.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libhashbrown.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libunwind.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libprofiler_builtins.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd_detect.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libpanic_abort.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring-core/android_arm64_armv8-2a_static_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring-test/android_arm64_armv8-2a_static_apex31/ -Z remap-cwd-prefix=. -C codegen-units=1 -C debuginfo=2 -C opt-level=3 -C relocation-model=pic -C overflow-checks=on -C force-unwind-tables=yes -C symbol-mangling-version=v0 --color always -A stable-features -Zdylib-lto   -C panic=abort -Z link-native-libraries=no -Z debug-info-for-profiling --cap-lints allow --edition=2018 -C metadata=libring --cfg 'feature="alloc"' --cfg 'feature="default"' --cfg 'feature="dev_urandom_fallback"' --cfg 'feature="once_cell"' --cfg 'feature="std"' -lstatic=ring-core -lstatic=ring-test --crate-type=rlib --crate-name=ring --target=aarch64-linux-android --sysroot=/dev/null && grep "^/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib:" /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d.raw > /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d                                                                                              error: failed to build archive: Unknown attribute kind (86) (Producer: 'LLVM17.0.2' Reader: 'LLVM 15.0.6-rust-dev')     error: aborting due to previous error

10:37:23 ninja failed with: exit status 1

**Comment by RT1648** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> FAILED: /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib OUT_DIR=out ANDROID_RUST_VERSION=1.68.0 CARGO_CRATE_NAME=ring CARGO_PKG_VERSION=0.17.0-alpha.11 AR=prebuilts/clang/host/linux-x86/clang-r487747c/bin/llvm-ar prebuilts/rust/linux-x86/1.68.0/bin/rustc -C linker=build/soong/scripts/mkcratersp.py --emit link -o /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib --emit dep-info=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d.raw external/rust/crates/ring/src/lib.rs --extern libc=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/libc/liblibc/android_arm64_armv8-2a_rlib_rlib-std_apex31/liblibc.rlib --extern once_cell=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/once_cell/libonce_cell/android_arm64_armv8-2a_rlib_rlib-std_apex31/libonce_cell.rlib --extern spin=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/spin/libspin/android_arm64_armv8-2a_rlib_rlib-std_apex31/libspin.rlib --extern untrusted=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/untrusted/libuntrusted/android_arm64_armv8-2a_rlib_rlib-std_apex31/libuntrusted.rlib --extern std=/home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd/android_arm64_armv8-2a_rlib_apex31/libstd.rlib -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcompiler_builtins.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcore.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/librustc_demangle.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/liblibc.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libcfg_if.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/liballoc.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libhashbrown.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libunwind.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libprofiler_builtins.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libstd_detect.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/prebuilts/rust/libpanic_abort.rust_sysroot/android_arm64_armv8-2a_rlib_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring-core/android_arm64_armv8-2a_static_apex31/ -L /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring-test/android_arm64_armv8-2a_static_apex31/ -Z remap-cwd-prefix=. -C codegen-units=1 -C debuginfo=2 -C opt-level=3 -C relocation-model=pic -C overflow-checks=on -C force-unwind-tables=yes -C symbol-mangling-version=v0 --color always -A stable-features -Zdylib-lto -C panic=abort -Z link-native-libraries=no -Z debug-info-for-profiling --cap-lints allow --edition=2018 -C metadata=libring --cfg 'feature="alloc"' --cfg 'feature="default"' --cfg 'feature="dev_urandom_fallback"' --cfg 'feature="once_cell"' --cfg 'feature="std"' -lstatic=ring-core -lstatic=ring-test --crate-type=rlib --crate-name=ring --target=aarch64-linux-android --sysroot=/dev/null && grep "^/home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib:" /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d.raw > /home/pixe/redwood/kartik/bo/out/soong/.intermediates/external/rust/crates/ring/libring/android_arm64_armv8-2a_rlib_rlib-std_apex31/libring.rlib.d error: failed to build archive: Unknown attribute kind (86) (Producer: 'LLVM17.0.2' Reader: 'LLVM 15.0.6-rust-dev') error: aborting due to previous error
> 
> 10:37:23 ninja failed with: exit status 1

I'm Facing Too On LineageOS, PixelExperience & PixelOS

**Reply by Thereache** [reply]
I am faced on bliss os

**Reply by eduardovicente** [reply]
I am facing this for crDroid

**Reply by RT1648** [reply]
Wait for sometime, source might be WIP

**Reply by Antony-Braiano** [reply]
Any fix for this?

**Reply by RT1648** [reply]
It's Fixed Long Ago... It Was A Source Side Error

**Reply by Thereache** [reply]
This error from server change your tamate ya restart your server

**Comment by jfullstackdev** [comment]
from Copilot : 

---

The error message suggests that there's a mismatch between the LLVM versions used to produce and read the Rust code. The producer is using LLVM 17.0.2, while the reader is using LLVM 15.0.6-rust-dev.

This kind of error can occur when you're trying to compile Rust code with a different version of the Rust compiler (rustc) than the one used to write the code. The Rust compiler uses LLVM as a backend, and different versions of rustc may use different versions of LLVM.

Here are a few potential solutions:

1. **Update your Rust compiler**: If you're using an older version of the Rust compiler, updating it might resolve the issue. You can update rustc using rustup:

```bash
rustup update
```

2. **Specify the Rust version**: If the code requires a specific Rust version, you can specify it using rustup. For example, if the code requires Rust 1.68.0, you can install it and set it as the default version like this:

```bash
rustup install 1.68.0
rustup default 1.68.0
```

3. **Check the LLVM version**: If updating or specifying the Rust version doesn't help, you might need to check the LLVM version used by your Rust compiler. You can do this by running `rustc --version --verbose` and looking at the LLVM version.

4. **Rebuild the project**: If you've made changes to the Rust or LLVM version, try rebuilding the project. Make sure to clean the build first to remove any artifacts from the previous build.

If none of these solutions work, you might need to seek help from the maintainers of the Rust code you're trying to compile. They might be able to provide more specific guidance based on the exact requirements of the code.

**Reply by Thereache** [reply, marked as answer]
Thanks sirji after changing the tamate and restart the server it will be fixed

**Reply by jfullstackdev** [reply]
maybe it is better to mark my actual answer than your reply so others can see the answer right away, thanks

**Comment by Thereache** [comment]
Thanks to help
